packages/manager/apps/pci-object-storage/src/pages/objects/container/object/versions/ShowVersions.page.tsx

```diff
@@ -0,0 +1,258 @@
+  const { storageId, objectName } = useParams();
+
+  const [searchParams] = useSearchParams();
+  const { data: project } = useProject();
+
+  const { hasMaintenance, maintenanceURL } = useProductMaintenance(
+    project?.project_id,
+  );
+
+  const { formatBytes } = useBytes();
+
+  const hrefProject = useProjectUrl('public-cloud');
+  const { t: tObjects } = useTranslation('objects');
+  const { t: tContainer } = useTranslation('container');
+
+  const objectStorageHref = useHref('..');
+  const containerDetailsHref = useHref(
+    `../${storageId}?region=${searchParams.get('region')}`,
+  );
+
+  const { data: region } = useGetRegion(
+    project?.project_id,
+    searchParams.get('region'),
+  );
+
+  const { storage: targetContainer, storages } = useStorage(
+    project?.project_id,
+    storageId,
+    searchParams.get('region'),
+  );
+
+  const { url } = useStorageEndpoint(project?.project_id, targetContainer);
+
+  const { data: serverContainer } = useServerContainer(
+    project?.project_id,
+    searchParams.get('region'),
+    targetContainer?.name,
+    targetContainer?.id,
+  );
+
+  const container = useMemo((): TContainer => {
+    if (!serverContainer) return undefined;
+    const s3StorageType = targetContainer?.s3StorageType;
+
+    return {
+      ...serverContainer,
+      id: serverContainer?.id || targetContainer?.id,
+      name: serverContainer?.name || targetContainer?.name,
+      objectsCount:
+        serverContainer?.storedObjects || serverContainer?.objectsCount,
+      usedSpace: formatBytes(
+        serverContainer?.storedBytes || serverContainer?.objectsSize,
+        2,
+        1024,
+      ),
+      publicUrl: url,
+      s3StorageType,
+      regionDetails: s3StorageType ? region : undefined,
+      staticUrl: serverContainer?.staticUrl || serverContainer?.virtualHost,
+    };
+  }, [serverContainer, region, targetContainer, url]);
+
+  const {
+    data: ObjectsVersions,
+    isLoading: isObjectsLoading,
+    refetch: refetchObjectVersions,
+    fetchNextPage,
+    hasNextPage,
+    isFetchingNextPage,
+  } = useServerContainerObjectVersions({
+    projectId: project?.project_id,
+    region: searchParams.get('region'),
+    name: storageId,
+    key: encodeURIComponent(objectName),
+    isS3StorageType: container?.s3StorageType,
+  });
+
+  useEffect(() => {
+    if (container?.s3StorageType) {
+      refetchObjectVersions();
+    }
+  }, [
+    storageId,
+    searchParams.get('region'),
+    storages,
+    container,
+    searchParams.get('refetch'),
+    refetchObjectVersions,
+  ]);
+
+  const handleFetchNextPage = () => {
+    if (hasNextPage && !isFetchingNextPage) {
+      fetchNextPage();
+    }
+  };
+
+  const containerObjectsWithIndex = useMemo(() => {
+    if (!ObjectsVersions && !container?.s3StorageType) return [];
+    return ObjectsVersions?.map((object, index) => ({
+      ...object,
+      index: `${index}`,
+    }));
+  }, [ObjectsVersions]);
+
+  const is = {
```

Comment:
I guess you can put the `useMemo` function at object level :)

```suggestion
  const is = useMemo(() =>  ({...
```